### PR TITLE
Implements Message model and relevant test cases.

### LINKED ISSUE #52: Implement messages model class
Implement messages model class

- message_hash (PK) (can be pre-calculated in facilitator.ts)
- gateway address
- type (“stake_and_mint” and “redeem_and_unstake”)
- source_status (Undeclared, Declared, Progressed, RevocationDeclared, Revoked)
- target_status (same)
- hash_lock
- secret
- gas_price
- gas_limit
- nonce
- sender (is staker_proxy, should be emitted in event from requestStake)
- direction (o2a, a2o)
- source_declaration_block_height

Abstract the database interaction layer.
(MessageRepository, Query… etc remember the discussion)

### PR DESCRIPTION
**Pull request by puneet-khushwani-eth**
Fixes #52 

1. Implemented Message model.
2. Added association between message & stake request.
3. Changed get method of stake request to always fetch message along with it.

### CODE REVIEW

File: src/models/MessageRepository.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    MessageConstant.revocationDeclaredStatus,
+    MessageConstant.revokedStatus,
+  ];
+
+  /* Origin to auxiliary direction. */
+  public static originToAuxiliaryDirection: string = 'o2a';
+
+  /* Auxiliary to origin direction. */
+  public static auxiliaryToOriginDirection: string = 'a2o';
+}
```
Comment (deepesh-kn) [inline src/models/MessageRepository.ts]: These all can be ENUMs
```
enum MessageType {
    Stake = 'stake',
    Redeem = 'redeem',
}
```    
```
enum MessageStatus {
    Undeclared = 'undeclared',
    Declared = 'declared',
    Progressed = 'progressed',
    RevocationDeclated = 'revocation_declared',
    Revoked = 'revoked',
}
```
```
enum MessageDirection {
    OriginToAuxuliary = 'o2a',
    AuxiliaryToOrigin = 'a2o',
}

```

File: src/models/MessageRepository.ts [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        },
+        sourceStatus: {
+          type: DataTypes.ENUM({
+            values: MessageConstant.statusesArray,
+          }),
+          allowNull: false,
+        },
+        targetStatus: {
+          type: DataTypes.ENUM({
+            values: MessageConstant.statusesArray,
```
Comment (deepesh-kn) [inline src/models/MessageRepository.ts]: Target status can never be `RevocationDeclared`.

File: src/models/MessageRepository.ts [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+          }),
+          allowNull: false,
+        },
+        gasPrice: {
+          type: DataTypes.INTEGER,
```
Comment (deepesh-kn) [inline src/models/MessageRepository.ts]: We should use `BIGINT`.
For gasLinit, nonce also

File: src/models/StakeRequestRepository.ts [hunk 4]
```diff
@@ -72,39 +78,48 @@ export class StakeRequestRepository {
...
             min: 0,
           },
         },
         beneficiary: {
           type: DataTypes.STRING,
+          allowNull: false,
         },
         gasPrice: {
           type: DataTypes.INTEGER,
+          allowNull: false,
```
Comment (deepesh-kn) [inline src/models/StakeRequestRepository.ts]: Thanks for adding this here 👍 
We missed checking that the datatype here is INTEGER and it should be BIGINT.

File: tests/models/MessageRepository/create.test.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    config = {
+      db: await Database.create(),
+    };
+  });
+
+  it('Checks creation of message model.', async (): Promise<void> => {
+    const messageAttributes: MessageAttributes = {
+      messageHash: '0x497A49648885f7aaC3d761817F191ee1AFAF399CFHKHFKDHKSDK343DDFDFD',
+      type: MessageConstant.stakeAndMintType,
+      gatewayAddress: '0x497A49648885f7aaC3d761817F191ee1AFAF399C',
```
Comment (deepesh-kn) [inline tests/models/MessageRepository/create.test.ts]: Please use addresses like `0x000000000000000000000000000000000000001`

File: src/models/MessageRepository.ts (hunk 1 above)
Comment (puneet-khushwani-eth) [reply src/models/MessageRepository.ts]: This is much better 👍. Have changed usages accordingly.

**Pull request merged**